contracts/roles/RevocationRegistryOnChain.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >= 0.7.0 < 0.9.0;
+
+import "@ensdomains/ens-contracts/contracts/registry/ENSRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "@ew-did-registry/proxyidentity/contracts/IOwned.sol";
+import "./RoleDefinitionResolver.sol";
+import {ClaimManager} from "./ClaimManager.sol";
+
+interface EthereumDIDRegistry {
+    function identityOwner(address identity) external view returns(address);
+    function validDelegate(address identity, bytes32 delegateType, address delegate) external view returns(bool);
+}
+    
+contract RevocationRegistryOnChain {
+    bytes32 constant private ASSERTION_DELEGATE_TYPE = 0x766572694b657900000000000000000000000000000000000000000000000000;
+    string constant private ERC712_DOMAIN_NAME = "Revocation Registry";
+    string constant private ERC712_DOMAIN_VERSION = "1.0";
```

Comment:
If the `ERC712` variables are present, is that implying that the contract should inherit from `ERC712`?

Reply:
I'm not sure if `ERC712` needs to be inherited for RevocationRegistry, I had the name and version defined by default. Would love to know your thoughts.